specifications/FiniteMonotonic/Finitize_CRDT.tla

```diff
@@ -0,0 +1,72 @@
+--------------------------- MODULE Finitize_CRDT ----------------------------
+
+EXTENDS Naturals
+
+CONSTANTS Node, Divergence
+
+VARIABLES counter, converge
+
+vars == <<counter, converge>>
+
+S == INSTANCE CRDT
+
+TypeOK ==
+  /\ S!TypeOK
+  /\ counter \in [Node -> [Node -> 0 .. Divergence]]
+  /\ converge \in BOOLEAN
+
+Safety == S!Safety
+
+Monotonicity == [][
+  \/ S!Monotonic
+  \/ \A a, b, c, d \in Node :
+    (counter'[a][b] - counter[a][b]) = (counter'[c][d] - counter[c][d])
+]_vars
+
+Liveness == converge ~> S!Convergence
+
+Init ==
+  /\ S!Init
+  /\ converge = FALSE
+
+Increment(n) ==
+  /\ ~converge
+  /\ counter[n][n] < Divergence
+  /\ S!Increment(n)
+  /\ UNCHANGED converge
+
+Gossip(n, o) ==
+  /\ S!Gossip(n, o)
+  /\ UNCHANGED converge
+
+Converge ==
+  /\ converge' = TRUE
+  /\ UNCHANGED counter
+
+GarbageCollect ==
```

Comment:
There is no difference between checking Finitize_CRDT.tla/Finitize_CRDT.cfg with or without the sub-action `GarbageCollect` present.

Reply:
That is true, but this is a demo - if you are building a system on top of this CRDT which does other actions that progress the state beyond touching the counters, this ensures the system does not deadlock and can evolve as long as the counters stay within the convergence limit.

Reply:
Also I believe it would explore the same state space but have one more state transition (so one more non-unique state, is how it would show up in the output).

Reply:
With `GarbageCollect`: 585,401 states generated, 50,000 distinct states found

Without `GarbageCollect`: 535,401 states generated, 50,000 distinct states found

However, I would expect the liveness property to fail. I will look into this!

Reply:
Oh yes, I see why the liveness property succeeds. Anyway this overall would support liveness properties being checked over a system that requires more steps than permitted by a simple state constraint to accomplish something.

Reply:
> With `GarbageCollect`: 585,401 states generated, 50,000 distinct states found
> 
> Without `GarbageCollect`: 535,401 states generated, 50,000 distinct states found
> 
> However, I would expect the liveness property to fail. I will look into this!

The "states generated" statistics is higher with `GarbageCollect` because there is one more sub-action for which TLC generates successor states.

Reply:
Also notice that `Liveness` holds even if `Fairness` is changed to `WF_vars(Next)`.  This is because `Liveness` is trivially true of behaviors with no `Converge` actions, and all other behavior guarantee that there are only finitely many `Increment` steps.

Reply:
> There is no difference between checking Finitize_CRDT.tla/Finitize_CRDT.cfg with or without the sub-action `GarbageCollect` present.

> Also notice that Liveness holds even if Fairness is changed to WF_vars(Next). This is because Liveness is trivially true of behaviors with no Converge actions, and all other behavior guarantee that there are only finitely many Increment steps. However, we would expect CRDT to violate its liveness property "eventually consistent" if fairness is asserted over Next.

Indeed, `GarbageCollect` does make a difference once the `converge` variable is eliminated (see https://github.com/tlaplus/Examples/pull/97#discussion_r1803666561).